### ISSUE TITLE
Support running a prebuilt apk with more than one launchable activity

### ISSUE THREAD

**Issue by mravn-google** [flutter/flutter#18451]
Create a Flutter/Android app with two main/launchable activities
```xml
        <activity
            android:name=".MainActivity"
            android:label="@string/app_name"
            android:theme="@style/AppTheme.NoActionBar">
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>
        <activity
            android:name=".MainActivity2"
            android:label="@string/app_name_2"
            android:enabled="false"
            android:theme="@style/AppTheme.NoActionBar">
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>
```
* Launching the app from Android Studio results in `MainActivity` being launched.
* Launching the app using `flutter run --use-application-binary` confusingly results in `MainActivity2` being launched.
* Disabling `MainActivity2` by changing `false` to `true` in the above xml results in nothing being launched using `flutter run --use-application-binary`

**Comment by mravn-google** [comment]
Proposed change by add2app early adopter:
```diff
--- a/packages/flutter_tools/lib/src/android/android_device.dart
+++ b/packages/flutter_tools/lib/src/android/android_device.dart
@@ -410,7 +410,8 @@ class AndroidDevice extends Device {
 
     cmd = adbCommandForDevice(<String>[
       'shell', 'am', 'start',
-      '-a', 'android.intent.action.RUN',
+      '-a', 'android.intent.action.MAIN',
+      '-c', 'android.intent.category.LAUNCHER',
       '-f', '0x20000000', // FLAG_ACTIVITY_SINGLE_TOP
       '--ez', 'enable-background-compilation', 'true',
       '--ez', 'enable-dart-profiling', 'true',
@@ -434,7 +435,9 @@ class AndroidDevice extends Device {
       if (debuggingOptions.useTestFonts)
         cmd.addAll(<String>['--ez', 'use-test-fonts', 'true']);
     }
-    cmd.add(apk.launchActivity);
+    final String packageName = apk.launchActivity.substring(0, apk.launchActivity.indexOf('/'));
+    printStatus('Launching activity ${packageName}');
+    cmd.add(packageName);
     final String result = (await runCheckedAsync(cmd)).stdout;
     // This invocation returns 0 even when it fails.
     if (result.contains('Error: ')) {
```

**Comment by mravn-google** [comment]
Android Studio appears to use the following command for launching the activity:
```
$ adb shell am start -n "io.flutter.mravn.using01/io.flutter.mravn.using01.MainActivity" -a android.intent.action.MAIN -c android.intent.category.LAUNCHER –D
```

**Comment by szakarias** [comment]
Fixed in #18716.
